### ISSUE TITLE
Can't get with_callback to work

### ISSUE THREAD

**Issue by andrewhavens** [nathankot/webstub#33]
I don't know if I'm doing something wrong, but I can't seem to get the `with_callback` assertions to work.

      stub = stub_request(:get, "http://api.example.com").
        to_return(json: {"hello" => "Hello!"})
      stub.with_callback do |headers, body|
          headers.should == 'foo' # should fail, but is never called
          body.should == 'bar'
          @request_headers = "foo"
          @request_body = "bar"
        end

      AFMotion::JSON.get 'http://api.example.com' do |response|
        @response = response
        resume
      end

      wait_max 1.0 do
        @request_headers.should == 'foo' # should pass, but var is nil
        @request_body.should == 'bar'
        @response.should.be.success
        @response.object.should == {"hello" => "Hello!"}
      end

The response is being returned correctly, but the `@request_headers` and `@request_body` are not being set. It's like the callback is never called.

**Comment by nathankot** [comment]
@andrewhavens have you tried `puts` inside `with_callback` to verify the callback is being called?
I haven't touched ruby in so long... does `@request_headers` inside `with_callback` have a lexical scope?

But why not just use it like this:

```ruby
  stub_request(:get, "http://api.example.com").
    to_return(json: {"hello" => "Hello!"}).
    with_callback do |headers, body|
      headers.should == 'foo'
      body.should == 'bar'
    end

  AFMotion::JSON.get 'http://api.example.com' do |response|
    response.should.be.success
    resume
  end
```

**Comment by andrewhavens** [comment]
Since the call to AFMotion::JSON.get is asynchronous, assertions within the response callback block are not going to work because the spec block will end before the callback gets called. You still need to apply a wait block. I'm trying to assert that the request is made with specific headers. I'm not sure how to do that and it seems that the with_callback block is not working as advertised.

**Comment by andrewhavens** [comment]
I looked into the code a little more. It looks like the callback is not called when there is no request body. https://github.com/nathankot/webstub/blob/a0651f09972f49c6b952b85e7bf2b71edad77b3a/lib/webstub/protocol.rb#L93-L95

**Comment by nathankot** [comment]
@andrewhavens Ahh yep you're right. I've fixed and pushed this - it's available in` v1.1.5` ;)
Thanks for filing the issue!
